src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -493,7 +525,144 @@ begin
   simp only [set.mem_to_finset, mem_neighbor_set],
 end
 
-/-! ## Subgraphs of subgraphs -/
+end subgraph
+
+section mk_properties
+/-! ### Properties of `singleton_subgraph` and `subgraph_of_adj` -/
+
+variables {G : simple_graph V} {G' : simple_graph W}
+
+instance nonempty_singleton_subgraph_verts (v : V) : nonempty (G.singleton_subgraph v).verts :=
+⟨⟨v, set.mem_singleton v⟩⟩
+
+@[simp] lemma singleton_subgraph_le_iff (v : V) (H : G.subgraph) :
+  G.singleton_subgraph v ≤ H ↔ v ∈ H.verts :=
+begin
+  refine ⟨λ h, h.1 (set.mem_singleton v), _⟩,
+  intro h,
+  split,
+  { simp [h] },
+  { simp [-set.bot_eq_empty] }
+end
+
+@[simp] lemma map_singleton_subgraph (f : G →g G') {v : V} :
+  subgraph.map f (G.singleton_subgraph v) = G'.singleton_subgraph (f v) :=
+by ext; simp only [relation.map, subgraph.map_adj, singleton_subgraph_adj, pi.bot_apply,
+  exists_and_distrib_left, and_iff_left_iff_imp, is_empty.forall_iff, subgraph.map_verts,
+  singleton_subgraph_verts, set.image_singleton]
+
+@[simp] lemma neighbor_set_singleton_subgraph (v w : V) :
+  (G.singleton_subgraph v).neighbor_set w = ∅ :=
+by { ext u, refl }
+
+@[simp] lemma edge_set_singleton_subgraph (v : V) :
+  (G.singleton_subgraph v).edge_set = ∅ :=
+begin
+  ext e,
+  refine e.ind _,
+  intros,
+  simp [-set.bot_eq_empty, Prop.bot_eq_false],
+end
+
+lemma eq_singleton_subgraph_iff_verts_eq (H : G.subgraph) {v : V} :
+  H = G.singleton_subgraph v ↔ H.verts = {v} :=
+begin
+  split,
+  { rintro rfl,
+    simp, },
+  { intro h,
+    ext,
+    { rw h,
+      simp, },
+    { simp only [Prop.bot_eq_false, singleton_subgraph_adj, pi.bot_apply, iff_false],
+      intro ha,
+      have ha1 := ha.fst_mem,
+      have ha2 := ha.snd_mem,
+      rw [h, set.mem_singleton_iff] at ha1 ha2,
+      subst_vars,
+      exact ha.ne rfl } },
+end
+
+instance nonempty_subgraph_of_adj_verts {v w : V} (hvw : G.adj v w) :
+  nonempty (G.subgraph_of_adj hvw).verts := ⟨⟨v, by simp⟩⟩
+
+@[simp] lemma edge_set_subgraph_of_adj {v w : V} (hvw : G.adj v w) :
+  (G.subgraph_of_adj hvw).edge_set = {⟦(v, w)⟧} :=
+begin
+  ext e,
+  refine e.ind _,
+  simp only [set.mem_singleton_iff, subgraph.mem_edge_set, subgraph_of_adj_adj, eq_comm,
+    iff_self, forall_2_true_iff],
```

Comment:
Will probably be shorter using `eq_singleton_iff_unique_mem`

Reply:
I couldn't get this one shorter. The `simp only` in this case is purely for speed.